File: src/modules/user/user.entity.ts
import {
  Entity,
  BaseEntity,
  PrimaryGeneratedColumn,
  Column,
  OneToMany,
  Unique,
  CreateDateColumn
} from "typeorm";
import * as bcrypt from "bcrypt";
import { Pet } from "src/modules/pet/pet.entity";

@Entity("User")
@Unique(["email"])
export class User extends BaseEntity {
  @PrimaryGeneratedColumn("uuid", { name: "id" })
  id: string;

  @Column()
  email: string;

  @Column({ name: "password", length: 200 })
  password: string;

  @Column({ name: "signature" })
  signature: string;

  @Column({ type: "bool", name: "isActive", default: true })
  isActive: boolean;

  @CreateDateColumn({
    type: "timestamp",
    name: "creation_date"
  })
  creationDate: string;

  @OneToMany(
    () => Pet,
    pet => pet.user
  )
  pets: Pet[];

  async validatePassword(password: string): Promise<boolean> {
    const hash = await bcrypt.hash(password, this.signature);
    return hash === this.password;
  }
}

File: src/main/java/dev/vitorvidal/petmanagementapi/model/entity/UserEntity.java
package dev.vitorvidal.petmanagementapi.model.entity;

import lombok.Getter;
import lombok.Setter;
import org.springframework.data.cassandra.core.mapping.*;

import javax.crypto.Mac;
import javax.crypto.spec.SecretKeySpec;
import java.nio.charset.StandardCharsets;
import java.time.LocalDateTime;
import java.util.Base64;
import java.util.Objects;
import java.util.UUID;

@Getter
@Setter
@Table(value = "user")
public class UserEntity {

    @PrimaryKey(value = "email")
    private String email;
    @Column
    private String username;
    @Column(value = "user_id")
    @Indexed
    private UUID userId;
    @Column
    private String password;
    @Column
    private String signature;
    @Column(value = "creation_date")
    @CassandraType(type = CassandraType.Name.TIMESTAMP)
    private LocalDateTime creationDate;
    @Column(value = "is_active")
    private Boolean isActive;

    public UserEntity() {
    }

    public UserEntity(String username, String email, String password) {
        this.userId = UUID.randomUUID();
        this.username = username;
        this.email = email;
        this.signature = UUID.randomUUID().toString().replace("-", "");
        this.isActive = true;
        this.creationDate = LocalDateTime.now();
        this.password = password;
    }
}
